xfd/cleware/AmpelControl_Travis_BuildSystem.py

```diff
@@ -0,0 +1,257 @@
+#!/usr/bin/python
+
+import os
+import sys
+import urllib2
+import time
+import subprocess
+from optparse import OptionParser
+from thread import start_new_thread
+import json
+
+
+class ampel_state:
+	def __init__(self):
+		# modes
+		self.off = 0
+		self.on = 1
+		self.flash = 2
+		# state
+		self.state = [self.off,self.off,self.off]
+
+	def set_state(self, state):
+		self.state = state
+
+	def get_state(self):
+		return self.state
+
+
+# color: (0=red, 1=yellow, 2=blue/green)
+# mode: (0=off, 1=on, 2=flash)
+# red: if build state Failure/Error for at least one repo
+# yellow flash: if at least one repo is Active
+# green: if all repo build state Success
+# all flash: if any of the build url is invalid/no internet connection
+def extract_ampel_state(state):
+	message = "Ampel is "
+	state_out = [0,0,0]
+	if "red_anime" in state:
+		message += "flashing red"
+		state_out[0] = 2
+	elif "red" in state:
+		message += "red"
+		state_out[0] = 1
+	else:
+		message += "no red"
+		state_out[0] = 0
+
+	message += ", "
+	if "yellow_anime" in state:
+		message += "flashing yellow"
+		state_out[1] = 2
+	elif "yellow" in state:
+		message += "yellow"
+		state_out[1] = 1
+	else:
+		message += "no yellow"
+		state_out[1] = 0
+
+	message += " and "
+	if "blue_anime" in state or "grey_anime" in state:
+		message += "flashing green"
+		state_out[2] = 2
+	elif "blue" in state:
+		message += "green"
+		state_out[2] = 1
+	else:
+		message += "no green"
+		state_out[2] = 0
+
+	if state_out == [0, 0, 0]:
+		message = "Error: invalid state. state = " + str(state)
+		state_out = [2,2,2]
+	return state_out, message
+
+def run_ampel():
+	global ampel
+	on = [False, False, False]
+	default_options = ["-c", "1"]
+	if options.device != None:
+		default_options += ["-d", str(options.device)]
+	while True:
+		state = ampel.get_state()
+		for color in [0,1,2]:
+			if state[color] == 2 and not on[color]:
+				p = subprocess.Popen(["clewarecontrol"] + default_options + ["-as", str(color), "1"], stderr=subprocess.STDOUT, stdout=subprocess.PIPE)
+				on[color] = True
+			elif state[color] == 2 and on[color]:
+				p = subprocess.Popen(["clewarecontrol"] + default_options + [ "-as", str(color), "0"], stderr=subprocess.STDOUT, stdout=subprocess.PIPE)
+				on[color] = False
+			else:
+				p = subprocess.Popen(["clewarecontrol"] + default_options + [ "-as", str(color), str(state[color])], stderr=subprocess.STDOUT, stdout=subprocess.PIPE)
+			p.wait()
+		time.sleep(1)
+
+
+def extract_build_query(file_path):
+	tag_store = []
+	with open(file_path) as fp:
+    		for line in fp:
+			if "git:" in line:
+				user = None
+        			repo = None
+        			branch = None
+                        elif "uri:" in line:
+				user = line.split("'https://github.com/",1)[1]
+				user = user.split("/", 1)[0].strip()
+                        elif "local-name:" in line:
+				repo = line.split(":",1)[1].strip()
+			elif "version:" in line:
```

Comment:
tabs vs. spaces: please use either tabs or spaces.

we generally prefer 4 spaces.